### PR TITLE
Implemented parameterized logging in Vertx core api.

### PR DESCRIPTION
**Pull request by manish-panwar**
Hey Guys,

I had opened a thread to discuss the support for parameterized logging in vertx - https://groups.google.com/forum/#!searchin/vertx/parameterized$20logging/vertx/R6DkZ23WJGU/2lBgvDb_JAEJ

I have made changes in logging api to support parameterized logging, please take a look and let me know if something looks off to you.

Thanks,
Manish

### CODE REVIEW

File: src/main/java/io/vertx/core/logging/impl/JULLogDelegate.java [hunk 1]
```diff
@@ -106,7 +156,12 @@ private void log(Level level, Object message, Throwable t) {
     // This will disable stack trace lookup inside JUL. If someone wants location info, they can use their own formatter
     // or use a different logging framework like sl4j, or log4j
     record.setSourceClassName(null);
+    record.setParameters(params);
 
     logger.log(record);
   }
+
+    private void log(Level level, Object message, Throwable t) {
+        log(level, message, t, null);
```
Comment (purplefox) [inline src/main/java/io/vertx/core/logging/impl/JULLogDelegate.java]: indentation.

please see the code style guidelines on the wiki

Comment (purplefox) [inline src/main/java/io/vertx/core/logging/impl/JULLogDelegate.java]: (we use two spaces not four)

**Comment by purplefox** [comment]
@manish-panwar could you please make sure you have signed the eclipse cla and the commits as described in CONTRIBUTING.MD and on the wiki, otherwise we can't accept the commits.

**Comment by purplefox** [comment]
Also please rebase with master :)

**Comment by purplefox** [comment]
@manish-panwar ping

**Comment by manish-panwar** [comment]
Sorry, I will do it today.

**Comment by purplefox** [comment]
Hi @manish-panwar - any updates on this.
We need to get this merged ASAP if it's going to make it into the 3.1 release

**Comment by purplefox** [comment]
Closing this, as user hasn't responded.

**Pull request closed without merging**